api/app/db/crud/hfi_calc.py

```diff
@@ -35,15 +35,23 @@ def get_fire_centre_stations(session, fire_centre_id: int) -> CursorResult:
 
 def get_most_recent_updated_hfi_request(session: Session,
                                         fire_centre_id: int,
-                                        prep_start_day: date = None,
-                                        prep_end_day: date = None) -> HFIRequest:
+                                        date_range: DateRange) -> HFIRequest:
     """ Get the most recently updated hfi request for a fire centre """
     query = session.query(HFIRequest)\
-        .filter(HFIRequest.fire_centre_id == fire_centre_id)
-    if prep_start_day is not None:
-        query = query.filter(HFIRequest.prep_start_day == prep_start_day)
-    if prep_end_day is not None:
-        query = query.filter(HFIRequest.prep_end_day == prep_end_day)
+        .filter(HFIRequest.fire_centre_id == fire_centre_id)\
+        .filter(HFIRequest.prep_start_day == date_range.start_date)\
+        .filter(HFIRequest.prep_end_day == date_range.end_date)
+    return query.order_by(HFIRequest.create_timestamp.desc()).first()
+
+
+def get_more_recent_updated_hfi_request_for_current_date(session: Session,
```

Comment:
Wouldn't `get_most_recent_updated_...` be a more accurate/descriptive name for this function?